File: Kendaraan Practice/Mobil.java
public class Mobil extends Kendaraan{
    private String supir;
    
    Mobil(){
        super();
        this.supir = "XXX";
    }

    Mobil(int nomor, int tahun, String merk, String supir)
    {
        super(nomor, tahun, merk);
        this.supir = supir;
    }
    
    Mobil(Mobil other)
    {
        super(other);
        this.supir = other.supir;
    }

    public Mobil copyUnit(Mobil other){
        super.copyUnit(other);
        this.supir = other.supir;
        return this;
    }

    @Override
    public void printInfo(){
        super.printInfo();
        System.out.printf("Kategori: Mobil%n");
        System.out.printf("Supir: %s%n", this.supir);
    }

    public int biayaSewa(int lamaSewa){
        return 500000*lamaSewa;
    }
}

File: Kendaraan Practice/Mobil.cpp
#include<bits/stdc++.h>
#include "Mobil.hpp"

using namespace std;

// Default ctor
Mobil::Mobil() : Kendaraan()
{
    this->supir = "XXX";
}

// User ctor
Mobil::Mobil(int nomor, int tahun, string merk, string supir) : Kendaraan(nomor, tahun, merk)
{
    this->supir = supir;
}

// cctor
Mobil::Mobil(const Mobil& other)
{
    this->nomor = other.nomor;
    this->tahun = other.tahun;
    this->merk = other.merk;
    this->supir = other.supir;
}

// dtor
Mobil::~Mobil() {}

void Mobil::printInfo()
{
    Kendaraan::printInfo();
    printf("Kategori: Mobil\n");
    printf("Supir\t: ");cout<< this->supir << "\n";
}

int Mobil::biayaSewa(int lamaSewa)
{
    return 5e5*lamaSewa;
}